### DISCUSSION TITLE
Link to other apps from a flatpak app

### DISCUSSION THREAD

**Discussion by selwynorren** [elementary/flatpak-platform#43] [Q&A] [answered]
Hi All, I am not really sure where to ask this, so here goes hoping its the right place.

So I am testing Elementary OS 6 and have the latest daily build updated. I have installed flathub as a system and not user and have gone ahead and installed filezilla. Everything works brilliantly, however, now I wanted to edit a remote file, and filezilla does not have an app associated with the file I want to edit (error_log). This is never normally a problem I just browse for io.elementary.code or any app that will server this purpose.

However what I can browse is extremely restricted. If I go look in /lib where io.elementary.code normally resides, from within filezilla, it does not show up at all. However if I navigate to it using files, then I can see it.

If I navigate to /var it only shows me filezilla itself. So it liek any flatpak app cannot navigate outside of its containerised eco system. I have the same issue with kdenlive, gimp, etc essentially any app installed s a flatpak and need to make use of an additional app to edit or open files.

So how do I go about instructing filezilla that I have Code or Sublime text or any other app for that matter?

**Comment by tintou** [comment, marked as answer]
Here is a link to the issue already reported upstream: https://github.com/flathub/org.filezillaproject.Filezilla/issues/40

**Reply by selwynorren** [reply]
Thanks @tintou Yeah this is not isolated to Filezilla, but any app that requires the use of another app (as I mentioned Kdenlive, Gimp, etc). I get the idea of sandboxing apps and i am all for it, however the barrier for entry to someone new to linux, just get harder when this stuff happens.

I would then mach rather stick to appimages of they are available, so I never have these same issues when using that tech, sadly not all app developers make appimages available.

**Comment by danirabbit** [comment]
Hey thanks for your question! This is intentional behavior. Flatpak apps are sandboxed so they can't see other apps on the host system. This prevents them from doing malicious things like executing other programs in the background or reading their files, or creating a profile of you based on the apps you have installed. This is all part of the push for a more secure and private system.

What apps can do however is use portals to interact with the host system in a secure and privacy respecting way. When apps want to open a file in another app they can use the "open uri" portal to either open the default app or show a dialog where you can choose which app to launch. Many apps are already updated to use portals, but some apps will need to be updated before they will work as expected.

As part of our policy in AppCenter, we require that apps published there be packaged and submitted by their actual developer. This ensures that the developer has tested the app to make sure you don't encounter issues like this. Other app stores, like FlatHub, don't necessarily have the same policy so your experience may vary. Whenever you sideload apps from an external source, you should expect that you may encounter these kinds of issues

**Reply by selwynorren** [reply]
Hey @danrabbit Thanks so much for the detailed answer, it really does shed a fair amount of light on the subject. If you don't mind me asking? Would this mean that app like Filezilla and kdenlive would have to change code on their side to accommodate flatpak? or would it be better to find alternate solutions to do the same work?

I ask this as I really love where elementary Os is going direction wise, However tools like kdenlive and filezilla, etc are a huge part of my workflow and I am pretty dependent on these tools for my day to day. So for now, while its in beta it does not matter to me, however once its released I am obviously concerned if everything is flatpak based and all sandboxed with no way around the basic functionality, I would rather test a few alternatives and see if they can replace these daily tools for me.

Obviously the other question that I have not yet asked is of these apps will still be able to be installed using deb packages like currently in 5.1.7. as that would be helpful

Really appreciate your time, thanks

**Reply by danirabbit** [reply]
Yup FileZilla and Kdenlive will need to make changes on their end to work as expected in a sandboxed environment. So installing them from Flatpak may not be viable right now if you depend on those features

You will indeed be able to sideload apps from any other method you would before like Deb packages, ppas, etc. Nothing restricts you from being able to install apps you want from the place you want :)

**Reply by selwynorren** [reply]
Hi @danrabbit Thanks so much for your time to answer, loved you livestream tonight, hope you rest well!!
